plugins/system/mvcoverride/helper/mvcloader.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * @package     RedCORE.Plugin
+ * @subpackage  System.MVCOverride
+ *
+ * @copyright   Copyright (C) 2008 - 2015 redCOMPONENT.com. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * Static class to handle loading of libraries.
+ *
+ * @since  1.4.10
+ */
+class MVCLoader extends JLoader
+{
+	protected static $changePrivate = 0;
+
+	protected static $prefix = '';
+
+	protected static $suffix = 'Default';
+
+	protected static $overrideFiles = array();
+
+	protected static $extendsClasses = array();
+
+	/**
+	 * Set in static array name class and relate data
+	 *
+	 * @param   string  $class           Name class
+	 * @param   string  $path            Path file
+	 * @param   bool    $isOverrideFile  Name class and path for override file
+	 * @param   string  $prefix          Prefix for extend files
+	 * @param   string  $suffix          Suffix for extend files
+	 *
+	 * @return void
+	 */
+	public static function setOverrideFile($class, $path, $isOverrideFile = false, $prefix = null, $suffix = null)
+	{
+		$object = new stdClass;
+		$object->path = $path;
+		$object->isOverride = $isOverrideFile;
+		$object->prefix = $prefix;
+		$object->suffix = $suffix;
+
+		if ($isOverrideFile)
+		{
+			$class = $prefix . $class . $suffix;
+		}
+
+		// Sanitize class name.
+		$class = strtolower($class);
+		self::$overrideFiles[$class] = $object;
+	}
+
+	/**
+	 * Method to setup the autoloaders for the Joomla Platform.
+	 *
+	 * @param   int     $changePrivate  Flag change values and functions in extends call to private instead protected
+	 * @param   string  $prefix         Prefix for extend files
+	 * @param   string  $suffix         Suffix for extend files
+	 *
+	 * @return  void
+	 */
+	public static function setupOverrideLoader($changePrivate = 0, $prefix = '', $suffix = 'Default')
+	{
+		self::$changePrivate = $changePrivate;
+		self::$prefix = $prefix;
+		self::$suffix = $suffix;
+
+		// Register the prefix autoloader.
+		spl_autoload_register(array('MVCLoader', '_override_load'), false, true);
+	}
+
+	/**
+	 * Autoload function for override files
+	 *
+	 * @param   string  $class  Name class search
+	 *
+	 * @return bool|mixed
+	 */
+	public static function _override_load($class)
```

Comment:
according to our convention this function name should be overrideLoad or maybe better loadOverride